src/SFA.DAS.AdminService.Web/Handlers/Gateway/GetAddressHandler.cs

```diff
@@ -0,0 +1,102 @@
+using SFA.DAS.AdminService.Web.Infrastructure;
+using SFA.DAS.AdminService.Web.ViewModels.Roatp.Gateway;
+using SFA.DAS.AssessorService.Application.Api.Client.Clients;
+using SFA.DAS.AssessorService.ApplyTypes.Roatp;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.AdminService.Web.Handlers.Gateway
+{
+    public class GetAddressHandler : IRequestHandler<GetAddressRequest, AddressCheckViewModel>
+    {
+        private readonly IRoatpApplicationApiClient _applyApiClient;
+        private readonly IQnaApiClient _qnaApiClient;
+        private readonly IRoatpApiClient _roatpApiClient;
+        private readonly ILogger<GetOrganisationStatusHandler> _logger;
+
+        public GetAddressHandler(IRoatpApplicationApiClient applyApiClient, IQnaApiClient qnaApiClient, IRoatpApiClient roatpApiClient, ILogger<GetOrganisationStatusHandler> logger)
+        {
+            _applyApiClient = applyApiClient;
+            _qnaApiClient = qnaApiClient;
+            _roatpApiClient = roatpApiClient;
+            _logger = logger;
+        }
+
+        public async Task<AddressCheckViewModel> Handle(GetAddressRequest request, CancellationToken cancellationToken)
+        {
+            var pageId = GatewayPageIds.AddressCheck;
+
+            var model = new AddressCheckViewModel { ApplicationId = request.ApplicationId, PageId = pageId };
+           
+            var currentRecord = await _applyApiClient.GetGatewayPageAnswer(request.ApplicationId, pageId);
+            var applicationDetails = await _applyApiClient.GetApplication(model.ApplicationId);
+
+            if (currentRecord?.GatewayPageData != null)
+            {
+                model = JsonConvert.DeserializeObject<AddressCheckViewModel>(currentRecord.GatewayPageData);
+                model.Status = currentRecord.Status;
+                model.Caption = RoatpGatewayConstants.Captions.OrganisationChecks;
+                model.Heading = RoatpGatewayConstants.Headings.AddressCheck;
+                return model;
+            }
+
+            if (applicationDetails?.ApplyData?.ApplyDetails?.ApplicationSubmittedOn != null)
+                model.ApplicationSubmittedOn = applicationDetails.ApplyData.ApplyDetails.ApplicationSubmittedOn;
+
+            model.Caption = RoatpGatewayConstants.Captions.OrganisationChecks;
+            model.Heading = RoatpGatewayConstants.Headings.AddressCheck;
+            model.SourcesCheckedOn = DateTime.Now;
+
+
+            var ukprn = await _qnaApiClient.GetQuestionTag(request.ApplicationId, RoatpQnaConstants.QnaQuestionTags.Ukprn);
+            model.Ukprn = ukprn;
+            model.UkrlpLegalName = await _qnaApiClient.GetQuestionTag(request.ApplicationId, RoatpQnaConstants.QnaQuestionTags.UKRLPLegalName);
+
+            var PreamblePage = await _qnaApiClient.GetPageBySectionNo(request.ApplicationId, 0, 1, RoatpQnaConstants.RoatpSections.Preamble.PageId);
```

Comment:
Replace magic numbers with constants

Reply:
We have discussed this with Mark. I don’t thing that there is real need for replacing sequenceNo/sectionNo, which clearly can be seen from the signature of the method with the very long wording RoatpQnaConstants.RoatpSections.SectionOne.

Reply:
I'm with @David-B-Read, I've no idea what the numbers mean.